src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -233,7 +238,36 @@ public sealed partial class Server : IDisposable {
         }
         #endregion
 
-        public void NotifyPackagesChanged(CancellationToken cancellationToken) {
+        public void HandlePathWatchChanges(bool watchSearchPaths) {
+            if (!watchSearchPaths) {
+                _searchPaths = null;
+                _watchSearchPaths = false;
+                _pathsWatcher?.Dispose();
+                _pathsWatcher = null;
+                return;
+            }
+
+            if (!_watchSearchPaths) {
+                _watchSearchPaths = true;
+                ResetPathWatcher();
+            }
+        }
+
+        private void ResetPathWatcher() {
+            if (!_watchSearchPaths) {
```

Comment:
Maybe consider taking this if statement out and put it where ResetPathWatcher is called